contributors/design-proposals/live-and-inplace-vertical-scaling.md

```diff
@@ -0,0 +1,325 @@
+Live and In-place Vertical Scaling Proposal
+================
+
+_Authors:_
+* @YoungjaeLee - Youngjae Lee &lt;leeyo@us.ibm.com&gt;
+* @karthickrajamani - Karthick Rajamani &lt;karthick@us.ibm.com&gt;
+
+# Abstract
+
+This proposal is to enable the resizing of resources allocated to running containers of a pod without having to restart the pod.
+The initial focus is on enabling this for Statefulsets, in particular, though this may be use-able for stand-alone/controller-free pods and Deployments as well.
+
+# Motivation
+
+Resources needed by containers can change over time for a variety of reasons - moving from live-test mode to production usage, change in user load or dataset sizes each of which again might come about for a variety of reasons.
+`Statefulset` supports the capability to change Request and Limit values specified for a container through supported pod spec update methods.
+However, they currently require the pods be restarted to run with the new resource sizes.
```

Comment:
IIRC, if change Requests and Limits values for a container of `Statefulset`, it will kill the old pod and create a new one with new resource requirement.

Reply:
Yes, the current Statefulset controller recreates pods if its resource allocation is changed. So, in this proposal, we modify the controller to leverage the proposed live resizing feature.